it/uk/gov/hmrc/excisemovementcontrolsystemapi/writes/GeneratedJsonWritersSpec.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.writes
+
+import generated.{IE704Type, IE801Type, IE802Type, IE803Type, IE807Type, IE810Type, IE813Type, IE815Type, IE818Type, IE819Type, IE829Type, IE837Type, IE839Type, IE840Type, IE871Type, IE881Type, IE905Type, XMLProtocol}
+import play.api.libs.json.OWrites
+import scalaxb.XMLFormat
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.messages.MessageTypeFormats.GeneratedJsonWriters
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import uk.gov.hmrc.excisemovementcontrolsystemapi.writes.testObjects._
+
+class GeneratedJsonWritersSpec extends AnyFreeSpec with GeneratedJsonWriters with Matchers {
+
+  "ie704Writes" - new TestType[IE704Type](IE704TestMessageType, this.IE704Type)
```

Comment:
We already have some Xml data for some message in /test/data directory. it would be nice to have these only in one place were we have all them without repeating the data. 

Also just a thought, I wonder if it would have been better testing the toJson method (which is probably what we are interested in) of each message instead of testing the implicit as integration test as the implicit method will be tested indirectly and may be a more complete integration test.